apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Outcome.java

```diff
@@ -0,0 +1,45 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2021 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.impl.transaction;
+
+public enum Outcome {
+    SUCCESS("success"),
+    FAILURE("failure"),
+    UNKNOWN("unknown");
+
+    /**
+     * String value used for serialization
+     */
+    private final String stringValue;
```

Comment:
store serialization value explicitly to avoid extra allocation and make it clear it should not be changed without having an impact elsewhere.